simtools/corsika/corsika_output.py

```diff
@@ -0,0 +1,1002 @@
+        """
+        Get the radial distribution of the photons on the ground in relation to the center of the
+        array.
+
+        Parameters
+        ----------
+        bin_size: float
+            Size of the step in distance (in meters).
+        max_dist: float
+            Maximum distance to consider in the 1D histogram (in meters).
+        density: bool
+            If True, returns the density distribution. If False, returns the distribution of counts.
+
+        Returns
+        -------
+        np.array
+            The edges of the 1D histogram in meters with size = int(max_dist/bin_size) + 1.
+        np.array
+            The values of the 1D histogram with size = int(max_dist/bin_size).
+        """
+        if self.telescope_indices is None:
+            if bin_size is None:
+                bin_size = 40
+            if max_dist is None:
+                max_dist = 1000
+        else:
+            if bin_size is None:
+                bin_size = 1
+            if max_dist is None:
+                max_dist = 10
+        edges_1D_list, hist1D_list = [], []
+        x_edges_list, y_edges_list, hist2D_values_list = self.get_2D_photon_position_distr(
+            density=density
+        )
+        for step, _ in enumerate(x_edges_list):
+            edges_1D, hist1D = convert_2D_to_radial_distr(
+                x_edges_list[step],
+                y_edges_list[step],
+                hist2D_values_list[step],
+                bin_size=bin_size,
+                max_dist=max_dist,
+            )
+            edges_1D_list.append(edges_1D)
+            hist1D_list.append(hist1D)
+        return np.array(edges_1D_list), np.array(hist1D_list)
+
+    def get_photon_wavelength_distr(self):
+        """
+        Get histograms with the wavelengths of the photon bunches.
+
+        Returns
+        -------
+        np.array
+            The edges of the wavelength histogram in nanometers.
+        np.array
+            The values of the wavelength histogram.
+        """
+        return self._get_1D("wavelength")
+
+    def get_photon_time_distr(self):
+        """
+        Get the distribution of the emitted time of the Cherenkov photons. The start of the time
+        is given according to the CORSIKA configuration.
+
+        Returns
+        -------
+        numpy.array
+            The edges of the direction histograms in ns.
+        numpy.ndarray
+            The values (counts) of the histogram.
+        """
+        return self._get_1D("time")
+
+    def get_photon_altitude_distr(self):
+        """
+        Get the emission altitude of the Cherenkov photons.
+
+        Returns
+        -------
+        numpy.array
+            The edges of the direction histograms in km.
+        numpy.ndarray
+            The values (counts) of the histogram.
+        """
+        return self._get_1D("altitude")
+
+    def get_num_photons_per_event_per_telescope(self):
```

Comment:
So, this is a more profound discussion. I turned this into a property for now but lets discuss it further.

I used many get functions in this class but they are actually not intended as properties but rather methods. would you suggest to convert all the get functions to pairs of propriety+setters or do you prefer me to change the name of the functions so that we dont have 'get_' at the beginning?

Reply:
I started thinking that we could remove some setters and put the code inside the properties themselves. We anyways dont expect many of the attributes to be altered, since they depend simply on the output, e.g. the position of telescopes, the position of the photons. what do you think?

Reply:
Some of the `get_` functions I didn't mind cause they were not related to an attribute and they involved actual calculations. However, this case seemed more trivial and it made sense to make it a property. Generally it's not a big deal to have a few "trivial" setters. 

I am not sure I understood the point about putting the code inside the properties.